plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -1369,4 +1376,26 @@ protected function number_format( $price, $order ) {
 
 		return number_format( (float) $price, $decimals, '.', '' );
 	}
+
+	/**
+	 * Maybe add or remove account restriction notice based on HTTP status code.
+	 *
+	 * Sets the account restriction flag if the status is 422 (account restricted),
+	 * or clears it if the status indicates success (200, 201).
+	 *
+	 * @param int   $http_code The HTTP status code from the PayPal API response.
+	 * @param array $response_data The response data from the PayPal API response.
+	 * @return void
+	 */
+	protected function maybe_add_or_remove_notice( $http_code, $response_data ) {
+		if ( in_array( $http_code, array( 200, 201 ), true ) ) {
```

Comment:
Nit: rather than requiring the notices class at the top of the file, can we require the file inline here when we need it?
```suggestion
		if ( ! class_exists( 'WC_Gateway_Paypal_Notices' ) ) {
			require_once __DIR__ . '/class-wc-gateway-paypal-notices.php';
		}

		if ( in_array( $http_code, array( 200, 201 ), true ) ) {
```